### DISCUSSION TITLE
How Can we use bazel for embedded system where toolchain is available on my PC.

### DISCUSSION THREAD

**Discussion by segfaultdev** [bazelbuild/bazel#24614] [Q&A]
I am trying to add tflite-micro to my project where I was using CMake earlier for building the project but then I realized that we need to use bazel to build the tflite-micro project so I am trying to migrate the whole project building process from CMake to Bazel but unfortunately I am unable to do so. currently I am lacking on toolchain configuration so that bazel use the `arm-none-eabi-gcc` and `arm-none-eabi-g++` instead of my PC's `gcc and g++` can anyone help me regarding that. I am totally new in bazel it would be a great help to me

**Comment by armandomontanez** [comment]
For the sake of completeness, I'll present two paths for creating a toolchain configuration:

## The new way: rules_cc rule-based toolchains
I just wrote a [blog post](https://pigweed.dev/docs/blog/06-better-cpp-toolchains.html#try-it-out) about the work I've done on this, importantly there's a few (very real!) examples towards the end that use the `arm-none-eabi` binutils suite. This approach is WAY easier to get started with, and is largely a usability redesign of the old mechanisms.

The [Raspberry Pi Pico's Bazel build](https://github.com/raspberrypi/pico-sdk/tree/95ea6acad131124694cda1c162c52cd30e0aece0/bazel/toolchain) uses these constructs.

## The old way: native Starlark primitives
The current "official" tutorial for declaring a C++ toolchain is [here](https://bazel.build/tutorials/ccp-toolchain-config), which I only list to acknowledge its existence. It's _extremely_ easy to trip over the constructs and use them wrong, so I'd advise against going down that route. In the fullness of time, I hope to rewrite that tutorial using the new way, but that likely won't happen for a while.
